### ISSUE TITLE
Inconsistent date field formatting in BibLaTeX export

### ISSUE THREAD

**Issue by shotcollin** [retorquere/zotero-better-bibtex#1493]
**Report ID:** 9VELWB5K-euc

**Exporter used:** Better BibLaTeX

**Expected behavior:** This is a bib entry categorized in Zotero as "Web Page" where Zotero fields `date` and `accessed` are both 4/10/2019. I haven't been able to locate any preference setting for date format, but I would expect both dates to be in the same format, whatever it is. For example:
```
@online{Beckford2019,
  title = {Blockchain Africa Conference 2019 Was a Success},
  author = {Beckford, Neil},
  date = {2019-10-04},
  journaltitle = {Cryptoline news},
  url = {www.cryptolinenews.com/industry-analysis/blockchain-africa-conference-2019},
  urldate = {2019-10-04},
  langid = {english}
}
```

**Actual behavior:** When exported as Better BibLaTeX, the entry's `date` and `urldate` are 2019-10-04 and 2019-04-10, respectively.

```
@online{Beckford2019,
  title = {Blockchain Africa Conference 2019 Was a Success},
  author = {Beckford, Neil},
  date = {2019-10-04},
  journaltitle = {Cryptoline news},
  url = {www.cryptolinenews.com/industry-analysis/blockchain-africa-conference-2019},
  urldate = {2019-04-10},
  langid = {english}
}
```
This causes the warning `Invalid field day for entrytype online` during `biber` data validation.

**Comment by blip-bloop** [comment]
:robot: this is your friendly neighborhood build bot announcing [test build 5.2.21.6388](https://github.com/retorquere/zotero-better-bibtex/releases/download/builds/zotero-better-bibtex-5.2.21.6388.xpi) ("add localeDateOrder to all tests")

Install in Zotero by downloading [test build 5.2.21.6388](https://github.com/retorquere/zotero-better-bibtex/releases/download/builds/zotero-better-bibtex-5.2.21.6388.xpi), opening the Zotero "Tools" menu, selecting "Add-ons", open the gear menu in the top right, and select "Install Add-on From File...".

**Comment by retorquere** [comment]
Can you get me a new debug log for that item using 6388?

**Comment by retorquere** [comment]
BTW, this is what that item looks like to me:

![Screenshot 2020-04-17 at 14 47 34](https://user-images.githubusercontent.com/132108/79570824-6ed5d000-80ba-11ea-9bd6-523079229357.png)

So it's probably a locale issue, but 6388 will tell me more.

**Comment by shotcollin** [comment]
Thanks for the quick response. New debug Report ID is 4JZQ7QY9-euc

Here's what the item looks like for me:

![BeckfordScreenshot](https://user-images.githubusercontent.com/19615630/79613830-54f1b880-80cd-11ea-8836-eca83f679119.png)

**Comment by blip-bloop** [comment]
:robot: this is your friendly neighborhood build bot announcing [test build 5.2.21.6398](https://github.com/retorquere/zotero-better-bibtex/releases/download/builds/zotero-better-bibtex-5.2.21.6398.xpi) ("fixes #1493")

Install in Zotero by downloading [test build 5.2.21.6398](https://github.com/retorquere/zotero-better-bibtex/releases/download/builds/zotero-better-bibtex-5.2.21.6398.xpi), opening the Zotero "Tools" menu, selecting "Add-ons", open the gear menu in the top right, and select "Install Add-on From File...".

**Comment by retorquere** [comment]
Thanks -- for reporting the issue, and confirming it was a locale problem. I've added a regression test that will prevent this from happening again -- 6398 has passed this test, but 6388 likely already fixed the problem for you.

The actual problem is that Zotero is making things pretty, thereby hiding the problem. If you click `Date`, it will say `4/10/2019`. If you click `Accessed`, it will say `2019-04-10`. Your locale is set to a place that for unfathomable reasons (sorry 😄 ) prefers `mdy` dates; the system that my tests run on defaults to `dmy`. `2019-04-10` unambiguously parses to 10th of April, 2019, but what `4/10/2019` means is locale-dependent, which is why on my system, the date looked different.

BBT will now use the current locale of Zotero to parse ambiguous dates, but keep in mind that if you sync this data to another system that has a different locale, the meaning of `4/10/2019` may change without notification. It is always preferable to use YYYY-MM-DD format because it will not show these kind of problems.

**Comment by blip-bloop** [comment]
:robot: this is your friendly neighborhood build bot announcing [test build 5.2.21.6399](https://github.com/retorquere/zotero-better-bibtex/releases/download/builds/zotero-better-bibtex-5.2.21.6399.xpi) ("fixes #1493")

Install in Zotero by downloading [test build 5.2.21.6399](https://github.com/retorquere/zotero-better-bibtex/releases/download/builds/zotero-better-bibtex-5.2.21.6399.xpi), opening the Zotero "Tools" menu, selecting "Add-ons", open the gear menu in the top right, and select "Install Add-on From File...".

**Comment by blip-bloop** [comment]
:robot: this is your friendly neighborhood build bot announcing [test build 5.2.21.6401](https://github.com/retorquere/zotero-better-bibtex/releases/download/builds/zotero-better-bibtex-5.2.21.6401.xpi) ("fixes #1493")

Install in Zotero by downloading [test build 5.2.21.6401](https://github.com/retorquere/zotero-better-bibtex/releases/download/builds/zotero-better-bibtex-5.2.21.6401.xpi), opening the Zotero "Tools" menu, selecting "Add-ons", open the gear menu in the top right, and select "Install Add-on From File...".
